src/transformation/visitors/conditional.ts

```diff
@@ -78,8 +80,15 @@ export function transformIfStatement(statement: ts.IfStatement, context: Transfo
 
     if (statement.elseStatement) {
         if (ts.isIfStatement(statement.elseStatement)) {
+            context.pushPrecedingStatements();
```

Comment:
I don't get why this is here. I assumed you were now forced to pop preceding statements whenever we visit a block, but then why is that not done on the `ifBlock` in this function?

Reply:
If an `else if` statement generates preceding statements, it needs to execute those in an else clause before doing the second conditional.
Example:
```ts
if (i) {
} else if (i) { // No preceding statements
}

if (i) {
} else if (i++) { // i++ creates preceding statements
}
```

```lua
if i then
elseif i then -- elseif is fine
end

if i then
else -- elseif won't work - we do an else, execute preceding statements, then do the if check
    local ____i_0 = i
    i = ____i_0 + 1
    if ____i_0 then
    end
end
```

I'll add some comments here explaining this.